libs/base/Data/List.idr

```diff
@@ -111,12 +111,49 @@ filter p (x :: xs)
         then x :: filter p xs
         else filter p xs
 
+public export
+filterWith : (a -> Bool) -> List a -> (List a, List Nat)
+filterWith p = h 0 where
+  h : Nat -> List a -> (List a, List Nat)
+  h _ [] = ([], [])
+  h lvl (x :: xs) = if p x
+      then let (ms, ns) = h (S lvl) xs in (x :: ms, lvl :: ns)
+      else h (S lvl) xs
+
+public export
+filterWith' : (a -> Bool) -> List a -> List (a, Nat)
+filterWith' p = h 0 where
+  h : Nat -> List a -> List (a, Nat)
+  h _ [] = []
+  h lvl (x :: xs) = if p x
+    then (x, lvl) :: h (S lvl) xs
+    else             h (S lvl) xs
+
```

Comment:
These seem like very ad-hoc function whose name is not very enlightening.
Other occurrences of `xWith` in the libs seem to be for functions of the shape:

```idris
xWith : Constraint b => (a -> b) -> f a -> g a
```

I think the name should be different & this should be in (at most) contrib.

Reply:
thx